vital/tests/test_geo_point.cxx

```diff
@@ -0,0 +1,224 @@
+IMPLEMENT_TEST(constructor_point)
+{
+  kwiver::vital::geo_point p{ loc1, crs_ll };
+
+  if ( p.is_empty() )
+  {
+    TEST_ERROR("The constructed point is empty");
+  }
+}
+
+// ----------------------------------------------------------------------------
+IMPLEMENT_TEST(assignment)
+{
+  kwiver::vital::geo_point p;
+  kwiver::vital::geo_point const p1{ loc1, crs_ll };
+  kwiver::vital::geo_point const p2;
+
+  if ( ! p.is_empty() )
+  {
+    TEST_ERROR("The default point is not empty");
+  }
+
+  p = p1;
+
+  if ( p.is_empty() )
+  {
+    TEST_ERROR("The point is empty after assignment from non-empty point");
+  }
+
+  if ( p.location() != p1.location() )
+  {
+    TEST_ERROR("The point has the wrong location after assignment");
+  }
+
+  if ( p.crs() != p1.crs() )
+  {
+    TEST_ERROR("The point has the wrong CRS after assignment");
+  }
+
+  p = p2;
+
+  if ( ! p.is_empty() )
+  {
+    TEST_ERROR("The point is not empty after assignment from empty point");
+  }
+}
+
+// ----------------------------------------------------------------------------
+IMPLEMENT_TEST(api)
+{
+  kwiver::vital::geo_point p{ loc1, crs_ll };
+
+  // Test values of the point as originally constructed
+  if ( p.location() != loc1 )
+  {
+    TEST_ERROR("The original location is incorrect");
+  }
+
+  if ( p.crs() != crs_ll )
+  {
+    TEST_ERROR("The original CRS is incorrect");
+  }
+
+  if ( p.location( crs_ll ) != loc1 )
+  {
+    TEST_ERROR("The location (requested in the original CRS) is incorrect");
+  }
+
+  // Modify the location
+  p.set_location( loc3, crs_utm_18n );
+
+  // Test the new values
+  if ( p.location() != loc3 )
+  {
+    TEST_ERROR("The original location is incorrect");
+  }
+
+  if ( p.crs() != crs_utm_18n )
+  {
+    TEST_ERROR("The original CRS is incorrect");
+  }
+
+  if ( p.location( crs_utm_18n ) != loc3 )
+  {
+    TEST_ERROR("The location (requested in the original CRS) is incorrect");
+  }
+
+  // Modify the location again
+  p.set_location( loc2, crs_ll );
+
+  // Test the new values
+  if ( p.location() != loc2 )
+  {
+    TEST_ERROR("The original location is incorrect");
+  }
+
+  if ( p.crs() != crs_ll )
+  {
+    TEST_ERROR("The original CRS is incorrect");
+  }
+
+  if ( p.location( crs_ll ) != loc2 )
+  {
+    TEST_ERROR("The location (requested in the original CRS) is incorrect");
+  }
+
+  // Test that the old location is not cached
+  try
+  {
+    if ( p.location( crs_utm_18n ) == loc3 )
+    {
+      TEST_ERROR("Changing the location did not clear the location cache");
+    }
+  }
+  catch (...)
+  {
+    // If no conversion functor is registered, the conversion will fail; that
+    // is okay, since we are only checking here that the point isn't still
+    // caching the old location, which it isn't if it needed to attempt a
+    // conversion
+  }
+}
+
+// ----------------------------------------------------------------------------
+IMPLEMENT_TEST(conversion)
```

Comment:
This test fails for me for both point and polygon with:

    Error: Unexpected exception: Failed to construct PROJ projection for EPSG:4326

I'll merge in release and push to see if I can get Travis CI to replicate that failure.